cupy/polynomial/polynomial.py

```diff
@@ -0,0 +1,27 @@
+import cupy
+
+
+def polyvander(x, deg):
+    """Computes the Vandermonde matrix of given degree.
+
+    Args:
+        x (cupy.ndarray): array of points
+        deg (int): degree of the resulting matrix.
+
+    Returns:
+        cupy.ndarray: The Vandermonde matrix
+
+    .. seealso:: :func:`numpy.polynomial.polynomial.polyvander`
+
+    """
+    deg.__index__()
+    if deg < 0:
+        raise ValueError('deg must be non-negative')
+    if not isinstance(x, cupy.ndarray) or x.ndim != 1:
+        x = cupy.array(x, copy=False, ndmin=1)
+    x = x + 0.0
```

Comment:
What is this?

Reply:
All types must have floating point entries in the output matrix

Reply:
Please cast with `x.astype(dtype, copy=False)`.

Reply:
casting with `x.astype(dtype, copy=False)` where dtype = `float` or `numpy.float32` or `numpy.float64` causes changes in floating point precision. Should I check whether the input is 32-bit or 64-bit float ?
I used `+ 0.0` for easy promotion between different precisions

Reply:
This typecast is not needed now.